lib/pbench/cli/agent/commands/results/move.py

```diff
@@ -0,0 +1,196 @@
+
+        runs_copied = 0
+        failures = 0
+        no_of_tb = 0
+
+        for dirent in self.config.pbench_run.iterdir():
+            if not dirent.is_dir():
+                continue
+            if dirent.name.startswith("tools-") or dirent.name == "tmp":
+                continue
+
+            no_of_tb += 1
+            result_dir = dirent
+
+            try:
+                mrt = MakeResultTb(result_dir, temp_dir, self.config, logger)
+            except FileNotFoundError as e:
+                logger.error("File Not Found Error, {}", e)
+                continue
+            except NotADirectoryError as e:
+                logger.error("Bad Directory, {}", e)
+                continue
+
+            try:
+                result_tb_name = mrt.make_result_tb()
+            except BadMDLogFormat as e:
+                logger.warning("Bad Metadata.log file encountered, {}", e)
+                failures += 1
+                continue
+            except FileNotFoundError as e:
+                logger.debug("File Not Found error, {}", e)
+                failures += 1
+                continue
+            except RuntimeError as e:
+                logger.warning("Unexpected Error encountered, {}", e)
+                failures += 1
+                continue
+            except Exception as e:
+                logger.debug("Unexpected Error occurred, {}", e)
+                failures += 1
+                continue
+
+            try:
+                crt = CopyResultTb(
+                    self.context.controller, result_tb_name, self.config, logger
+                )
+            except FileNotFoundError as e:
+                logger.error("File Not Found error, {}", e)
+                failures += 1
+                continue
+
+            try:
+                crt.copy_result_tb(self.context.token)
+            except (FileUploadError, RuntimeError) as e:
+                logger.error("Error uploading a file, {}", e)
+                failures += 1
+                continue
+
+            try:
+                # We always remove the constructed tar ball, regardless of success
+                # or failure, since we keep the result directory below on failure.
+                os.remove(result_tb_name)
+            except OSError:
+                logger.error("Failed to remove {}", result_tb_name)
+                failures += 1
+                continue
+
+            if copy_only:
+                copied = result_dir.parent / f"{result_dir.name}.copied"
+                try:
+                    copied.touch()
+                except OSError:
+                    logger.error("Failed to `touch` {}", copied)
+                    failures += 1
+                    continue
+            else:
+                try:
+                    shutil.rmtree(result_dir)
+                except OSError:
+                    logger.error(
+                        "Failed to remove the {} directory hierarchy", result_dir
+                    )
```

Comment:
This message is not reporting _why_ it failed (i.e., it should be attaching a stringized rendition of the exception object).

Reply:
Existing code.

Reply:
Needs improvement.

:-)